plugins/out_s3/s3.c

```diff
@@ -51,13 +51,14 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
 
 static int put_all_chunks(struct flb_s3 *ctx);
 
-static void cb_s3_upload(struct flb_config *ctx, void *data);
+static void cb_s3_upload(struct flb_config *ctx, void *data, time_t *file_first_log_time);
 
 static struct multipart_upload *get_upload(struct flb_s3 *ctx,
                                            const char *tag, int tag_len);
 
 static struct multipart_upload *create_upload(struct flb_s3 *ctx,
-                                              const char *tag, int tag_len);
+                                              const char *tag, int tag_len,
+                                              time_t *file_first_log_time);
```

Comment:
minor: why did you choose to use a time_t pointer? time_t is just an integer type. Normally if you were passing an int into these functions, you'd pass it by value, not reference.

Reply:
Oh sorry wait, I just realized, its a pointer because of the case where we don't have the time and so we use NULL for that? Which is probably better than using `0` to indicate we don't have the time. Makes sense.